qa/L0_perf_resnet/run_test.sh

```diff
@@ -73,22 +66,41 @@ fi
 
 set +e
 
-# Run the model once to warm up. Some frameworks do optimization on the first requests.
-# Must warmup similar to actual run so that all instances are ready
-$PERF_CLIENT -v -i ${PERF_CLIENT_PROTOCOL} -m $MODEL_NAME -p${MEASUREMENT_WINDOW} \
-                -b${STATIC_BATCH} --concurrency-range ${CONCURRENCY}
+# Overload use of PERF_CLIENT_PROTOCOL for convenience with existing test and 
+# reporting structure, though "triton_c_api" is not strictly a "protocol".
+if [[ "${PERF_CLIENT_PROTOCOL}" == "triton_c_api" ]]; then
+    # Server will be run in-process with C API
+    SERVICE_ARGS="--service-kind triton_c_api \
+                  --triton-server-directory ${TRITON_DIR} \
+                  --model-repository ${MODEL_REPO}"
+else
+    SERVICE_ARGS="-i ${PERF_CLIENT_PROTOCOL}"
+
+    SERVER_LOG="${NAME}.serverlog"
+    run_server
+    if (( $SERVER_PID == 0 )); then
+        echo -e "\n***\n*** Failed to start $SERVER\n***"
+        cat $SERVER_LOG
+        exit 1
+    fi
+
+    # Run the model once to warm up. Some frameworks do optimization on the first requests.
+    # Must warmup similar to actual run so that all instances are ready
+    # Note: Running extra PA for warmup doesn't make sense for C API since it
```

Comment:
Does the warmup cause significant difference in the performance of C-API models? If yes, maybe we can use the [server warmup](https://github.com/triton-inference-server/server/blob/main/docs/model_configuration.md#model-warmup) to make sure the perf numbers are accurate?

Reply:
Followup in other thread: https://github.com/triton-inference-server/server/pull/4420#discussion_r881866156